src/twister2/device/native_simulator_adapter.py

```diff
@@ -88,7 +88,7 @@ async def _run_command(self, command: list[str], timeout: float = 60.):
 
     def _run_simulation(self, build_dir: str | Path, timeout: float) -> None:
         command: list[str] = self._get_command(build_dir)
-        logger.info('Running command: %s', command)
+        logger.info('Running command: %s', ' '.join(command))
```

Comment:
Please consider coping into `src/twister2/helper.py` file following method: 
https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/twister/twisterlib/log_helper.py#L13-L27
Then it can be used in place like there for printing commands. `shlex` module is much more appropriate for printing commands than simple `' '.join(command)`. For example it detects when element on `command` list should be used with quotes `""`.

But using `shlex` has one minus - it not always works well on Windows OS, what I described here:
https://github.com/zephyrproject-rtos/zephyr/pull/49367#issuecomment-1224194528. So, that's why it is important to distinguish those two situations: working on Linux and on Windows and then applying `shlex` only on Linux.

Reply:
Updated.